crates/nu-protocol/src/locale.rs

```diff
@@ -0,0 +1,171 @@
+use std::{fmt, str::FromStr};
+use thiserror::Error;
+
+// TODO: this should be reworked. Currently, this contains both a `chrono::Locale` and a
+// `num_format::Locale` so that the same `Locale` type can be used to format numbers and dates.
+// In the future, we should possibly look into using `ICU4X` via the `icu` crate(s) to unify
+// the locale types as well as for better Intl.
+#[derive(Clone, Copy, PartialEq, Eq, Hash)]
+pub struct Locale {
+    number: num_format::Locale,
+    date: chrono::Locale,
+}
+
+#[allow(dead_code)]
+fn env(key: &'static str) -> Option<String> {
+    std::env::var(key).ok().filter(|s| !s.is_empty())
```

Comment:
Are `$env` entries synchronized with the process environment? If I set `$env.LC_TIME = "de_CH"`, will this function start returning it?

I think it's important to be able to set the current locale from within the nu script / repl itself.